src/data/two_pointing.lean

```diff
@@ -0,0 +1,92 @@
+/-
+Copyright (c) 2022 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import data.sum.basic
+import logic.nontrivial
+
+/-!
+# Two-pointings
+
+This file defines `two_pointing α`, the type of two pointings of `α`. A two-pointing is the data of
+two distinct terms.
```

Comment:
I think it might be helpful to briefly mention it's a kind of cospan in the category of sets. (I guess cospans under the terminal object with this distinctness property?) At least for me, it gives a sense of what the point of this module might be, but maybe that's just because you linked that nlab page.